Adjusting MyoBack model and Full body model

cherylwang20 opened the pull request:
Changes based on comments in this pr: [gh pr checkout 37](https://github.com/MyoHub/myo_sim/pull/37)

1. Change all mesh filenames to lowercase to align with the code standards: https://github.com/MyoHub/myo_sim/pull/38/commits/a730668110c6edb287ea9d3c0db7dcb5ee6ce0ee
2. Remove meshes/ribcage_s.stl, use thorax.stl: [cd6efca](https://github.com/MyoHub/myo_sim/pull/38/commits/cd6efca02dfb4d6b72bfcfd0e4402ae9c1730408)
3. eliminate internal collision of back/myoTorso_exosuit.xml and back/myoTorso.xml models have internal collisions. [078ba0a](https://github.com/MyoHub/myo_sim/pull/38/commits/078ba0a134c91b79ef12b4499dee428497b0964c) and [ab975b2](https://github.com/MyoHub/myo_sim/pull/38/commits/ab975b260eb12df86381262933e5e8eeecf4effa)
![image](https://github.com/user-attachments/assets/47dc99c4-3187-4f98-93b1-fbdbba221132)
![image](https://github.com/user-attachments/assets/4b77504a-9188-43fd-8e8f-f7c2fbab692d)
 
4. Previous keyframe issue - <key qpos='-0.025 0.1 1 1 0 0 0 0 0 0 0 0 0 0.1 0.1 0.7 0.08395 -0.02023 0.0130636 0.0430074 0.008183 0.02969 0.1048 0.039303 0.0226632 -0.0787215 0.0032808 -0.010751 -0.5236 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0'/>: this is only cause by actively adjusting the abs, which is meant to control passively with joints. I have lower the limit of the abs to decrease extreme behavior: [cd6efca](https://github.com/MyoHub/myo_sim/pull/38/commits/cd6efca02dfb4d6b72bfcfd0e4402ae9c1730408)
5. Missing lumbar1, this is a bug based which the geom is not displayed: [6523246](https://github.com/MyoHub/myo_sim/pull/38/commits/6523246d90ba5de4816e0bc7e2beb25fe9674348). Fixed:
![image](https://github.com/user-attachments/assets/150f6b7e-4244-47af-8683-5060b1d2bb95)

cherylwang20 commented:
@vikashplus @Vittorio-Caggiano

vikashplus commented on back/assets/myoTorso_assets.xml:
```diff
@@ -41,7 +41,7 @@
     <mesh name="torso_geom_10_thoracic3_s" file="../myo_sim/meshes/thoracic3_s.stl" scale="0.87 0.87 0.87"/>
     <mesh name="torso_geom_11_thoracic2_s" file="../myo_sim/meshes/thoracic2_s.stl" scale="0.87 0.87 0.87"/>
     <mesh name="torso_geom_12_thoracic1_s" file="../myo_sim/meshes/thoracic1_s.stl" scale="0.87 0.87 0.87"/>
-    <mesh name="torso_geom_13_ribcage_s" file="../myo_sim/meshes/ribcage_s.stl" scale="0.87 0.87 0.87"/>
```
If we added the ribcage_s.stl mesh. Perhaps we should remove it as well to avoid any future issues.

vikashplus commented on back/assets/myoTorso_body.xml:
```diff
@@ -176,7 +176,7 @@
         <joint name="LB_wrapjnt_t2" pos="0 0 0" axis="0 1 0" limited="true" type="slide" range="-.1 .1"/>
         <joint name="LB_wrapjnt_r3" pos="0 0 0" axis="0 0 1" limited="true" range="-.1 .1"/>
-        <geom name="Cylinder2_cylinder" size="0.07 0.13" pos="-0.118 0.0768 0" type="cylinder" group="2" rgba="0.19 0.83 0.78 0.2"/>
+        <geom name="Cylinder2_cylinder" contype="2" conaffinity="2" size="0.07 0.13" pos="-0.118 0.0768 0" type="cylinder" group="2" rgba="0.19 0.83 0.78 0.2"/>
```
Is this geom merely a wrapping object? 
Can you help me understand why -- contype="2" conaffinity="2" ?

cherylwang20 replied:
Hi Vikash, thanks for letting me know, I have created a separate wrapping class for all of these wrapping object: [e664b58](https://github.com/MyoHub/myo_sim/pull/38/commits/e664b58a4c8285deeaa254ab77b88db32ec53be9)

vikashplus commented on back/assets/myoTorso_body_exosuit.xml:
```diff
@@ -507,10 +507,10 @@
+                  <geom name="upper_exo_geom" contype="2" conaffinity="0" type="mesh" rgba="0.4 0.4 0.4 1" euler="1.57 3.14 1.57" group = "2" pos="-0.01 0.165 0" mesh="upper_exo_geom_1"/>
+                  <geom name="torso_geom_13" pos="0.028 0.365 0" type="mesh" rgba="1 1 1 1" mesh="torso_geom_13_ribcage_s"/>
+                  <geom name="Cylinder_cylinder" contype="2" conaffinity="2" size="0.17 0.13" pos="-0.21 0.04 0" type="cylinder" group="2" rgba="0.19 0.83 0.78 0.2"/>
+                  <geom name="Exo_cylinder" contype="2" conaffinity="2" size="0.015 0.13" pos="-0.07 0.06 0" type="cylinder" group="3" rgba="0 0 1 0.1"/>
```
contype="2" conaffinity="2" is not as common. Can you elaborate what you are trying to implement?

cherylwang20 replied:
I have created a separate wrapping class for all of these wrapping object: [e664b58](https://github.com/MyoHub/myo_sim/pull/38/commits/e664b58a4c8285deeaa254ab77b88db32ec53be9) and [268fc1f](https://github.com/MyoHub/myo_sim/pull/38/commits/268fc1faa8d3a4ded305946c148c4a4c921e1a0c). contype="2" conaffinity="2" is now removed.

vikashplus reviewed (changes requested):
I gave a quick browse. @Vittorio-Caggiano can you please take a pass on the biomechanical details? Please tag me back, and I'll make a final pass on the model. Note that my final pass will be very disruptive. This model is very monolithic and verbose. I'll be creating default classes and aggressively using them to simplify the model definition

cherylwang20 commented on back/assets/myoTorso_assets.xml:
```diff
@@ -41,7 +41,7 @@
     <mesh name="torso_geom_11_thoracic2_s" file="../myo_sim/meshes/thoracic2_s.stl" scale="0.87 0.87 0.87"/>
     <mesh name="torso_geom_12_thoracic1_s" file="../myo_sim/meshes/thoracic1_s.stl" scale="0.87 0.87 0.87"/>
-    <mesh name="torso_geom_13_ribcage_s" file="../myo_sim/meshes/ribcage_s.stl" scale="0.87 0.87 0.87"/>
+    <mesh name="torso_geom_13_ribcage_s" file="../myo_sim/meshes/thorax.stl" scale="1 1 1"/>
```
https://github.com/MyoHub/myo_sim/pull/38/files/ab975b260eb12df86381262933e5e8eeecf4effa#r2065386503

@vikashplus Hi Vikash, in the latest push, ribcage_s.stl is already removed, is it what you have in minid?

cherylwang20 commented:
Thanks for the review. The latest update includes:

1. Changed all wrapping objects to the same class of group and color. 
2. Removed redundant wrapping class in the chest. 
3. Clean up exo geom classes.

vikashplus commented:
<img width="556" alt="image" src="https://github.com/user-attachments/assets/14840b6b-fd22-493b-b0c5-71f4c727de1f" />

1. `Abdomen_ellipsoid` looks unnecessary. I haven't seen any reference to it anywhere 
2. Rename `Cylinder2_cylinder` and `Cylinder_cylinder` with more meaningful names

vikashplus commented:
better align `myoTorso_body_exosuit` and `myoTorso_body`. Diff the two files, and you will see that there are multiple conflicting details between the two models.

I think the best way to do this is to create a new model, exo_suite, and then add that on top of the existing body using constraints

vikashplus commented:
There are a lot of `sidesite` sites that are floating around in free space. 
<img width="434" alt="image" src="https://github.com/user-attachments/assets/db8a3cbb-3f2e-47af-b716-253e5125e5ac" />
It will be great to add a class for these as well, and move them to a group that is hidden by default

cherylwang20 commented:
> <img alt="image" width="556" src="https://private-user-images.githubusercontent.com/12837145/442918587-14840b6b-fd22-493b-b0c5-71f4c727de1f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDc1ODQwMzQsIm5iZiI6MTc0NzU4MzczNCwicGF0aCI6Ii8xMjgzNzE0NS80NDI5MTg1ODctMTQ4NDBiNmItZmQyMi00OTNiLWIwYzUtNzFmNGM3MjdkZTFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTE4VDE1NTUzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE2OTgyMDAwYjUxYWNjYWQ0ZjUyMzFkNWZhYTU3ODVlM2IyNTIwMGNjZTM2YzM5Yzc2NmE4ZmE1YzU4ZjUxMTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._s9kkGuKYDZOvXTLlOpXzxpFZH0hz7LstCuslJpRqM8">
> 1. `Abdomen_ellipsoid` looks unnecessary. I haven't seen any reference to it anywhere
> 2. Rename `Cylinder2_cylinder` and `Cylinder_cylinder` with more meaningful names,
remove the abdomen_ellipsoid geom, and rename Cylinder2_cylinder and Cylinder_cylinder to back_cylinder_s and back_cylinder_l accordingly.  [1437a9b](https://github.com/MyoHub/myo_sim/pull/38/commits/1437a9b08208b0806cfb08af0d3ba420e2ad1ad4)

cherylwang20 commented:
> 7ad2e17

Created new class for all sidesite, with group 3. [7ad2e17](https://github.com/MyoHub/myo_sim/pull/38/commits/7ad2e17eabfa87e93483404c1b15ce5134eb6491)

cherylwang20 commented:
> There are a lot of `sidesite` sites that are floating around in free space. <img alt="image" width="434" src="https://private-user-images.githubusercontent.com/12837145/442924850-db8a3cbb-3f2e-47af-b716-253e5125e5ac.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDc1ODkzODIsIm5iZiI6MTc0NzU4OTA4MiwicGF0aCI6Ii8xMjgzNzE0NS80NDI5MjQ4NTAtZGI4YTNjYmItM2YyZS00N2FmLWI3MTYtMjUzZTUxMjVlNWFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTE4VDE3MjQ0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc4YTI4ZjVhZDI1N2MzNmJiNGQ4ZjRmMGIzYTkzMTllNGNiZjhkNjQzNzczNTVmOGUxMjg1MzI0Y2ZmMWRkZmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.18j8udSg2aDK1NRMPrfgWVidsMbJz_mcxrUUTxnZGaQ"> It will be great to add a class for these as well, and move them to a group that is hidden by default
 
Created a new class for all sidesite, with group 3. [7ad2e17](https://github.com/MyoHub/myo_sim/pull/38/commits/7ad2e17eabfa87e93483404c1b15ce5134eb6491)

cherylwang20 commented:
> better align `myoTorso_body_exosuit` and `myoTorso_body`. Diff the two files, and you will see that there are multiple conflicting details between the two models.
> 
> I think the best way to do this is to create a new model, exo_suite, and then add that on top of the existing body using constraints

Synced the differences here (one joint range difference): [1631984](https://github.com/MyoHub/myo_sim/pull/38/commits/163198462fc44428e6fe1dc989ba80515804c310). 

Let me know if a new model is still necessary

vikashplus commented:
> > better align `myoTorso_body_exosuit` and `myoTorso_body`. Diff the two files, and you will see that there are multiple conflicting details between the two models.
> > I think the best way to do this is to create a new model, exo_suite, and then add that on top of the existing body using constraints
> 
> Synced the differences here (one joint range difference): [1631984](https://github.com/MyoHub/myo_sim/pull/38/commits/163198462fc44428e6fe1dc989ba80515804c310).
> 
> Let me know if a new model is still necessary

i think, it will be best to create the model. Any form of duplication leads to lots of problems down the road to keep things in sync.

cherylwang20 commented:
Latest update:

1. Both myoTorso and myoTorso_exosuit uses default myoTorso_body as main body.
2. Body meshes are hard to separate when creating a separate file, "exosuit". Directly inserting bodies in the myoTorso_exosuit allows the freedom to define free joints at will. 
3. Merge the assets file with and without the exosuit. The only difference is having three additional mesh geoms and an exomesh class definition.
4. Change the collision mesh of myofullbody to group 4.

@vikashplus looping back for review

vikashplus commented on back/myoTorso_exosuit.xml:
```diff
@@ -7,15 +7,49 @@
+        <include file="assets/myoTorso_body.xml"/>
+    </body>
+
+    <body name="exo_1" pos="-.025 0.1 0">
```
There seems to be a dummy body here.

cherylwang20 replied:
deleted here: [250a00d](https://github.com/MyoHub/myo_sim/pull/38/commits/250a00d63c7f61f49f1aae7ecf01ed10797d2c5a)

vikashplus commented on back/myoTorso_exosuit.xml:
```diff
@@ -7,15 +7,49 @@
+  <equality>
+    <weld body1="torso" body2="exo_torso"/>
+    <connect site1="Exo_RightShoulder" site2="RightShoulder"/>
+    <connect site1="Exo_LeftShoulder" site2="LeftShoulder"/>
```
I don't quite understand the approach here. Can you please explain. 
- If we have a weld, why do we need connect? 
- And if the idea is to use connect to find attachment points to the body, then why use weld at all? Just add more connect points, no?

cherylwang20 replied:
removed the connection equalities: [39db5bb](https://github.com/MyoHub/myo_sim/pull/38/commits/39db5bbd36318bd4638b90e1df1a26742d141bc9). The weld is now fixed to the chest location while motion remains intact from previous versions

vikashplus commented on back/myoTorso_exosuit.xml:
```diff
@@ -37,8 +71,8 @@
+  
   <keyframe>
-    <key name="default-pose" qpos='0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0'/>
+    <key name="default-pose" qpos='0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 -0.1051 0.091 1.05 0.707035 0.707179 0 0 -0.129 0.0975 1.05 0.707035 0.707179 0 0'/>
```
Any particular relevance of this keyframe? Seems quite close to the default. Should we just remove it?

cherylwang20 replied:
deleted here: [250a00d](https://github.com/MyoHub/myo_sim/pull/38/commits/250a00d63c7f61f49f1aae7ecf01ed10797d2c5a)

vikashplus reviewed (changes requested):
See comments. 
It will be good to also pay attention to formatting. There are spaces / tabs / indentation mismatch throughout
